ui/desktop/src/components/ChatView.tsx

```diff
@@ -69,6 +71,27 @@ const isUserMessage = (message: Message): boolean => {
   return true;
 };
 
+const substituteParameters = (prompt: string, params: Record<string, string>): string => {
+  let substitutedPrompt = prompt;
+
+  console.log('Substituting parameters:', { prompt, params });
+
+  for (const key in params) {
+    // Escape special characters in the key (parameter) and match optional whitespace
+    const regex = new RegExp(`{{\\s*${key.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')}\\s*}}`, 'g');
+    const beforeSubstitution = substitutedPrompt;
+    substitutedPrompt = substitutedPrompt.replace(regex, params[key]);
```

Comment:
Treat the parameter as literal text, to avoid regex injection and unintended matching.

For example, without santization
```js
// matches {{user_name}}, {{user-name}}, etc
substitutedPrompt.replace(new RegExp('{{user.name}}'), value) 
```
with sanitization
```js
// matches exactly {{user.name}}
substitutedPrompt.replace(new RegExp('{{user\.name}}'), value)
```